tests/unit/test_serialization.py

```diff
@@ -267,105 +375,167 @@ def assert_almost_equal(value: np.ndarray, reference: Any):
         np.testing.assert_almost_equal(value, reference)
 
 
-def assert_scenario_correct(
-    deserialized_dataset: Mapping[ComponentType, np.ndarray],
+def assert_single_dataset_entries(
+    deserialized_dataset: SingleDataset,
     serialized_dataset: Mapping[str, Any],
     sparse_components: list[ComponentType],
+    data_filter,
 ):
+    """Check a SingleDataset's individual components for correctness"""
+    # TODO What is the purpose of this check? Implement without using derserialized_dataset
     for key in serialized_dataset["data"]:
-        if key not in deserialized_dataset:
+        if key not in deserialized_dataset and not is_filtered_out(data_filter, key):
             assert key in sparse_components
```

Comment:
Weird check. I would assume that this is to check that if a componenet is filtered out, then it must live somewhere (`sparse_components`), but if you check that it is not in the `deserialized_dataset` but it was not filtered out either, why would it be in `sparse_components` then? Is this check actually being made?

Reply:
Exactly. I removed the check now. In the main as well it never made it to this control loop